server/backends/blobstore/blobstore.go

```diff
@@ -505,7 +511,7 @@ func (a *AwsS3BlobStore) createBucketIfNotExists(ctx context.Context, bucketName
 
 func (a *AwsS3BlobStore) ReadBlob(ctx context.Context, blobName string) ([]byte, error) {
 	start := time.Now()
-	b, err := a.download(ctx, blobName)
+	b, err := a.download(ctx, blobPath(blobName))
```

Comment:
nit: in the download func below, rename `blobName` to `path`. Or (probably cleaner): revert the change on this line, and instead change line 525 to `Key: aws.String(blobPath(blobName))`

Reply:
Agree! Changing line 525 made it looks a lot cleaner